src/features/island/collectibles/components/pets/PetFeed.tsx

```diff
@@ -0,0 +1,263 @@
+import React, { memo, useState } from "react";
+import { getPetRequestXP, Pet, PetName } from "features/game/types/pets";
+import { SplitScreenView } from "components/ui/SplitScreenView";
+import { Box } from "components/ui/Box";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { CookableName } from "features/game/types/consumables";
+import { Label } from "components/ui/Label";
+import { getExperienceToNextLevel } from "features/game/types/pets";
+import { Button } from "components/ui/Button";
+import { Inventory } from "features/game/types/game";
+import levelUp from "assets/icons/level_up.png";
+import { SUNNYSIDE } from "assets/sunnyside";
+import { InnerPanel } from "components/ui/Panel";
+import xpIcon from "assets/icons/xp.png";
+
+interface Props {
+  petName: PetName;
+  petData: Pet;
+  inventory: Inventory;
+}
+
+const PetFeedComponent: React.FC<
+  Props & {
+    handleFeed: (food: CookableName) => void;
+  }
+> = ({ petName, petData, handleFeed, inventory }) => {
+  const [selectedFood, setSelectedFood] = useState<CookableName>(
+    petData.requests.food[0],
```

Comment:
When no food exists, an error is thrown. Looks like this happens when a pet is first placed.

Consider handling cases where there are no food requests.

1. Place a new pet
2. Open up the Pet modal (before saving) = error state